src/server/main_service.cc

```diff
@@ -859,6 +860,15 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
     return true;
   });
 
+  config_registry.RegisterMutable("serialization_max_chunk_size",
+                                  [](const absl::CommandLineFlag& flag) {
```

Comment:
This is dangerous and also wrong because of MemoryBytesFlag. What Roman wants is to cache the value of the absl::flag either as a thread local or as a global variable which you set during Service::Init (see how we set server_state) and (IMO I prefer the first because of NUMA but it won't make a difference here.

P.s. reviewing code from my phone is not fun